src/Provider/AbstractProvider.php

```diff
@@ -349,14 +349,20 @@ protected function fetchProviderData($url, array $headers = [])
             $response = $request->getBody();
         } catch (BadResponseException $e) {
             // @codeCoverageIgnoreStart
-            $raw_response = explode("\n", $e->getResponse());
-            throw new IDPException(end($raw_response));
+            $message = $this->getResponseErrorMessage($e->getResponse()->getBody());
+            throw new IDPException($message);
```

Comment:
Actually, on second look... I'm not sure how `IDPException` is ever supposed to work correctly as written, because this code is assuming that the response is plaintext, but [IDPException is expecting an array](https://github.com/thephpleague/oauth2-client/blob/9f2725a336d3aec025b578204938db934a12a25d/src/Exception/IDPException.php#L13-L23).